sorts/sorts_test.go

```diff
@@ -1,112 +1,63 @@
 package sorts
 
-import "testing"
+import (
+	"fmt"
+	"testing"
+)
 
-//BEGIN TESTS
-
-func TestBubble(t *testing.T) {
+func testFramework(t *testing.T, sortingFunction func([]int) []int) {
 	for _, test := range sortTests {
-		actual := bubbleSort(test.input)
-		pos, sorted := compareSlices(actual, test.expected)
-		if !sorted {
-			if pos == -1 {
-				t.Errorf("test %s failed due to slice length changing", test.name)
+		t.Run(test.name, func(t *testing.T) {
+			actual := sortingFunction(test.input)
+			pos, sorted := compareSlices(actual, test.expected)
```

Comment:
I checked just now and the performance on both of them is similar. I think that the compareSlices function is better for this task since it provides more information